pallets/file-system/src/utils.rs

```diff
@@ -294,4 +416,39 @@ where
         );
         block_to_insert_expiration
     }
+
+    /// Challenge `file_key` to force the BSP to update its storage root to uninclude the
+    /// file and reduce the data used by the storage provider.
+    fn challenge_and_reduce_data_used(
```

Comment:
There probably isn't much need for this function since we shouldn't challenge when a BSP stops storing a file.